### DISCUSSION TITLE
Encryption seems not wot work in Isar v4

### DISCUSSION THREAD

**Discussion by abhay-s-rawat** [isar/isar#1358] [Help]
I am not sure how encryption is working but I read in documentation that encryption only works when engine is sqlite.

Step 1: I created a database and inserted 1000 records with some encyption key.
Step2: I am reading same database but with diffrent key but instead of giving error it is giving correct record, seems like encryption is not working.

```
final Isar isar = await Isar.openAsync(
      schemas: [IsarUserSchema],
      directory: dir.path,
      engine: IsarEngine.sqlite,
      encryptionKey: "testKey",
    );
```

**Comment by abhay-s-rawat** [comment]
Plus if I execute below, they both give opposite bool values.
isar.close() always return false
when I run isar.isOpen after closing it will return isar.isOpen=false
So I don't know it is open or closed.
```
bool isClosed = isar.close();
print( 'isar object closed isClosed: $isClosed, isar.isOpen: ${isar.isOpen}');
```

**Reply by Flajt** [reply]
This first thing checks if it's closed, which it is so `isClosed = true` and the second one checks if it's open, which it should not be, since you closed it, which means `isOpen = false`. If I don't misunderstand anything both should be different values and not the same, if they were the same you would have a problem because it would say: db is open and db is closed concurrently.

**Reply by abhay-s-rawat** [reply]
It was a bug which got fixed https://github.com/isar/isar/issues/1360
